src/Webhook/Mappers/AbstractMapper.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+
+namespace TrackMage\WordPress\Webhook\Mappers;
+
+use TrackMage\WordPress\Exception\InvalidArgumentException;
+use TrackMage\WordPress\Exception\EndpointException;
+use TrackMage\WordPress\Repository\ShipmentRepository;
+
+class AbstractMapper implements EntityMapperInterface {
+
+
+    protected $data;
+    protected $updatedFields;
+    protected $requestBody;
+    protected $repo;
+
+    protected $source;
+    protected $entity;
+
+    protected $map = [];
+
+    protected $workspace;
+
+    public function __construct($source = null) {
+        $this->workspace = get_option( 'trackmage_workspace' );
+        $this->source = $source;
+    }
+
+    /**
+     * @param array $item
+     *
+     * @return bool
+     */
+    public function supports( array $item ) {
+        return false;
+    }
+
+    /**
+     * @param array $item
+     */
+    public function handle( array $item ) {
+        // do nothing in abstract
+    }
+
+
+    /**
+     * Check if entity from TrackMage can be handled
+     *
+     * @return bool
+     */
+    protected function canHandle(){
```

Comment:
maybe rename `canHandle` to `validateData`? It throws exception if data is wrong